From 5b0d33bb20d633e5aa92fa17c4784e2024afaea6 Mon Sep 17 00:00:00 2001
From: Dudemanguy <random342@airmail.cc>
Date: Wed, 17 Jul 2024 20:23:56 -0500
Subject: [PATCH] build: remove systemd dependency and code

autotools is too painful to bother doing better so just hack it in
---
 configure.ac                                  | 4 ----
 modules/action-button/gp-lock-screen-applet.c | 5 ++---
 modules/launcher/gp-launcher-applet.c         | 5 ++---
 modules/menu/gp-menu-utils.c                  | 5 ++---
 4 files changed, 6 insertions(+), 13 deletions(-)

diff --git a/configure.ac b/configure.ac
index 8945b92df..094bc5f68 100644
--- a/configure.ac
+++ b/configure.ac
@@ -130,7 +130,6 @@ LIBWNCK_REQUIRED=43.0
 LIBGNOME_MENU_REQUIRED=3.7.90
 LIBECAL_REQUIRED=3.33.2
 LIBEDATASERVER_REQUIRED=3.5.3
-LIBSYSTEMD_REQUIRED=230
 CAIRO_REQUIRED=1.0.0
 GWEATHER_REQUIRED=3.91.0
 DCONF_REQUIRED=0.13.4
@@ -169,7 +168,6 @@ PKG_CHECK_MODULES([ACTION_BUTTON], [
   gnome-desktop-3.0 >= $LIBGNOME_DESKTOP_REQUIRED
   gtk+-3.0 >= $GTK_REQUIRED
   libgnome-menu-3.0 >= $LIBGNOME_MENU_REQUIRED
-  libsystemd >= $LIBSYSTEMD_REQUIRED
   x11
 ])
 
@@ -182,7 +180,6 @@ PKG_CHECK_MODULES([LAUNCHER], [
   gnome-desktop-3.0 >= $LIBGNOME_DESKTOP_REQUIRED
   gtk+-3.0 >= $GTK_REQUIRED
   libgnome-menu-3.0 >= $LIBGNOME_MENU_REQUIRED
-  libsystemd >= $LIBSYSTEMD_REQUIRED
 ])
 
 PKG_CHECK_MODULES([MENU], [
@@ -191,7 +188,6 @@ PKG_CHECK_MODULES([MENU], [
   gnome-desktop-3.0 >= $LIBGNOME_DESKTOP_REQUIRED
   gtk+-3.0 >= $GTK_REQUIRED
   libgnome-menu-3.0 >= $LIBGNOME_MENU_REQUIRED
-  libsystemd >= $LIBSYSTEMD_REQUIRED
 ])
 
 PKG_CHECK_MODULES([NOTIFICATION_AREA], [
diff --git a/modules/action-button/gp-lock-screen-applet.c b/modules/action-button/gp-lock-screen-applet.c
index 14ce595c3..350b9491b 100644
--- a/modules/action-button/gp-lock-screen-applet.c
+++ b/modules/action-button/gp-lock-screen-applet.c
@@ -20,7 +20,6 @@
 
 #include <gio/gdesktopappinfo.h>
 #include <glib/gi18n-lib.h>
-#include <systemd/sd-journal.h>
 
 #include "gpab-screensaver-gen.h"
 
@@ -209,7 +208,7 @@ screensaver_cb (GSimpleAction *action,
 static void
 child_setup (gpointer user_data)
 {
-  GAppInfo *info;
+/*  GAppInfo *info;
   const gchar *id;
   gint stdout_fd;
   gint stderr_fd;
@@ -229,7 +228,7 @@ child_setup (gpointer user_data)
     {
       dup2 (stderr_fd, STDERR_FILENO);
       close (stderr_fd);
-    }
+    }*/
 }
 
 static void
diff --git a/modules/launcher/gp-launcher-applet.c b/modules/launcher/gp-launcher-applet.c
index bf42f7226..328165e9c 100644
--- a/modules/launcher/gp-launcher-applet.c
+++ b/modules/launcher/gp-launcher-applet.c
@@ -33,7 +33,6 @@
 #include <glib/gi18n-lib.h>
 #include <gmenu-tree.h>
 #include <libgnome-desktop/gnome-systemd.h>
-#include <systemd/sd-journal.h>
 
 #include "gp-launcher-button.h"
 #include "gp-launcher-properties.h"
@@ -758,7 +757,7 @@ launch_animation (GpLauncherApplet *self)
 static void
 child_setup (gpointer user_data)
 {
-  GAppInfo *info;
+/*  GAppInfo *info;
   const gchar *id;
   gint stdout_fd;
   gint stderr_fd;
@@ -778,7 +777,7 @@ child_setup (gpointer user_data)
     {
       dup2 (stderr_fd, STDERR_FILENO);
       close (stderr_fd);
-    }
+    }*/
 }
 
 static void
diff --git a/modules/menu/gp-menu-utils.c b/modules/menu/gp-menu-utils.c
index 65bc5bb24..f8d19549c 100644
--- a/modules/menu/gp-menu-utils.c
+++ b/modules/menu/gp-menu-utils.c
@@ -20,7 +20,6 @@
 #include <glib/gi18n-lib.h>
 #include <gtk/gtk.h>
 #include <libgnome-desktop/gnome-systemd.h>
-#include <systemd/sd-journal.h>
 
 #include "gp-menu-utils.h"
 
@@ -262,7 +261,7 @@ get_label_if_file (GFile *file)
 static void
 child_setup (gpointer user_data)
 {
-  GAppInfo *info;
+/*  GAppInfo *info;
   const gchar *id;
   gint stdout_fd;
   gint stderr_fd;
@@ -282,7 +281,7 @@ child_setup (gpointer user_data)
     {
       dup2 (stderr_fd, STDERR_FILENO);
       close (stderr_fd);
-    }
+    }*/
 }
 
 static void
-- 
2.45.2

